test/Vault.spec.ts

```diff
@@ -1272,6 +1302,22 @@ describe('Vault', () => {
   });
 
   describe('deposit', () => {
+    it('revers with an error when the underlying has fees', async () => {
+      await underlying.setFee(500);
+
+      await addUnderlyingBalance(alice, '1000');
+
+      const tx = vault.connect(alice).deposit(
+        depositParams.build({
+          amount: parseUnits('500'),
+          inputToken: underlying.address,
+          claims: [claimParams.percent(100).to(alice.address).build()],
+        }),
+      );
+
+      await expect(tx).to.be.revertedWith('VaultAmountDoesNotMatchParams');
```

Comment:
maybe minor nitpicking, but could this error be more descriptive, telling something about underlying with fees is not supported?

Reply:
I guess the cause can be something else other than that. You can always see the test to find an instance of when this error can happen.

Reply:
Ok, I'm fine with that